src/i_input.c

```diff
@@ -782,31 +899,70 @@ void I_HandleKeyboardEvent(SDL_Event *sdlevent)
     {
         case SDL_KEYDOWN:
             event.type = ev_keydown;
-            event.data1.i = TranslateKey(sdlevent->key.keysym.scancode);
+            event.data1.i = TranslateKey(&sdlevent->key.keysym);
+            event.data2.i = GetLocalizedKey(&sdlevent->key.keysym);
 
             if (event.data1.i != 0)
             {
                 D_PostEvent(&event);
             }
+
+            if (text_input_enabled && vanilla_keyboard_mapping)
+            {
+                event.type = ev_text;
+                event.data1.i = GetTypedChar(&sdlevent->key.keysym);
+
+                if (event.data1.i != 0)
+                {
+                    D_PostEvent(&event);
+                }
+            }
             break;
 
         case SDL_KEYUP:
             event.type = ev_keyup;
-            event.data1.i = TranslateKey(sdlevent->key.keysym.scancode);
+            event.data1.i = TranslateKey(&sdlevent->key.keysym);
 
-            // data2/data3 are initialized to zero for ev_keyup.
-            // For ev_keydown it's the shifted Unicode character
-            // that was typed, but if something wants to detect
-            // key releases it should do so based on data1
+            // data2 is initialized to zero for ev_keyup. For ev_keydown it's
+            // the shifted Unicode character that was typed, but if something
+            // wants to detect key releases it should do so based on data1
             // (key ID), not the printable char.
 
+            event.data2.i = 0;
+
             if (event.data1.i != 0)
             {
                 D_PostEvent(&event);
             }
             break;
 
+        case SDL_TEXTINPUT:
+            if (text_input_enabled
+                && strlen(sdlevent->text.text) == 1
+                && (sdlevent->text.text[0] & 0x80) == 0)
+            {
+                event.type = ev_text;
+                event.data1.i = sdlevent->text.text[0];
+                D_PostEvent(&event);
+            }
+            break;
+
         default:
             break;
     }
 }
+
+void I_InitKeyboard(void)
+{
+    // On desktop platforms, SDL_StartTextInput() is implicitly called on SDL
+    // window creation
+    if (vanilla_keyboard_mapping)
```

Comment:
I think this should always be disabled at startup, regardless of vanilla keyboard mapping.

Reply:
Done in https://github.com/fabiangreffrath/woof/pull/2030/commits/99a1c70e631d965978a00184107dca98df0f9638